### PR TITLE
polish python support

### PR DESCRIPTION
**Pull request by goulvenclech**
Polish Python support.

## What has changed?

- `pip.rs`: dependency updates now preserve extras, environment markers, and mixed naming conventions; duplicate package names are rejected; complex constraints are skipped (See #175 ). Also use proper glob matching for `[tool.uv.workspace]` members, add `exclude` pattern support.

## How is this tested?

- `pip_tests.rs`: 26 new unit tests covering normalization edge cases, all PEP 508 operators, extras/markers preservation, and collision detection scenarios.

## How is this documented?

N/A

### CODE REVIEW

File: crates/sampo-core/src/adapters/pypi/pip.rs [hunk 1]
```diff
@@ -381,25 +426,80 @@ fn parse_project_metadata(source: &str) -> ProjectMetadata {
...
+    let Some(workspace) = doc
+        .get("tool")
+        .and_then(|t| t.as_table())
+        .and_then(|t| t.get("uv"))
+        .and_then(|t| t.as_table())
+        .and_then(|t| t.get("workspace"))
+        .and_then(|t| t.as_table())
+    else {
+        return UvWorkspaceConfig::default();
+    };
```
Comment (rafaeelaudibert) [inline crates/sampo-core/src/adapters/pypi/pip.rs]: I don't know enough about rust, but don't you need to cast this to `UvWorkspaceConfig` on the `Some` case to make sure it's always a `UvWorkspaceConfig`? Will Rust be smart enough to do that based on the `else` clause?

**Review by rafaeelaudibert — APPROVED** [review]
Great improvements! We don't have any of those complex usecases on posthog hence why I didn't add them, but it's obviously good to support them all!
